zeppelin/hdi_add_zeppelin.sh

```diff
@@ -0,0 +1,220 @@
+
+create_config()
+{
+    curl -u admin:$AMBARI_PASSWORD -i -X POST -d '{"type": "'"$1"'", "tag": "install", "properties": '"$2"'}' https://$CLUSTER.azurehdinsight.net/api/v1/clusters/$CLUSTER/configurations -H "X-Requested-By:ambari"
+}
+
+#----------------------------------------------------------------
+# ADD CONFIG
+#----------------------------------------------------------------
+apply_config()
+{
+
+    curl -u admin:$AMBARI_PASSWORD -i -X PUT -d '{"Clusters": {"desired_configs": { "type": "'"$1"'", "tag" :"install" }}}' https://$CLUSTER.azurehdinsight.net/api/v1/clusters/$CLUSTER -H "X-Requested-By:ambari"
+}
+
+#----------------------------------------------------------------
+# INSTALL
+#----------------------------------------------------------------
+install()
+{
+    curl -u admin:$AMBARI_PASSWORD -i -X PUT -d '{"ServiceInfo": {"state" : "INSTALLED"}}'  https://$CLUSTER.azurehdinsight.net/api/v1/clusters/$CLUSTER/services/ZEPPELIN -H "X-Requested-By:ambari"
+    wait_until "INSTALLED"
+}
+
+start_service()
+{
+    curl -u admin:$AMBARI_PASSWORD -i -X PUT -d '{"ServiceInfo": {"state" : "STARTED"}}'  https://$CLUSTER.azurehdinsight.net/api/v1/clusters/$CLUSTER/services/ZEPPELIN -H "X-Requested-By:ambari"
+    wait_until "STARTED"
+}
+
+#----------------------------------------------------------------
+# WAIT UNTIL
+#----------------------------------------------------------------
+
+wait_until()
+{
+    finished=0
+    while [ $finished -ne 1 ]
+    do
+      str=$(curl -s -u admin:$AMBARI_PASSWORD https://$CLUSTER.azurehdinsight.net/api/v1/clusters/$CLUSTER/services/ZEPPELIN)
+      if [[ $str == *"$1"* ]] || [[ $str == *"Service not found"* ]] 
+      then
+        finished=1
+      fi
+      sleep 3
+    done
+}
+#----------------------------------------------------------------
+# MAIN
+#----------------------------------------------------------------
+process_arguments $@
+
+validate_arguments
+
+validate_ambari_credentials
+
+curl -u admin:$AMBARI_PASSWORD -i -X POST -d '{"ServiceInfo":{"service_name":"ZEPPELIN"}}' https://$CLUSTER.azurehdinsight.net/api/v1/clusters/$CLUSTER/services -H "X-Requested-By:ambari"
+
+curl -u admin:$AMBARI_PASSWORD -i -X POST -d '' https://$CLUSTER.azurehdinsight.net/api/v1/clusters/$CLUSTER/services/ZEPPELIN/components/ZEPPELIN_MASTER -H "X-Requested-By:ambari"
+create_config zeppelin-config "$ZEPPELIN_CONFIG"
+
+create_config zeppelin-env "$ZEPPELIN_ENV"
+
+create_config zeppelin-shiro-ini "$ZEPPELIN_SHIRO_INI"
+
+create_config zeppelin-log4j-properties "$ZEPPELIN_LOG4J_PROPERTIES"
+
+apply_config zeppelin-config
+
+apply_config zeppelin-env
+
+apply_config zeppelin-shiro-ini
+
+apply_config zeppelin-log4j-properties
+
+curl -u admin:$AMBARI_PASSWORD -i -X POST -d '{"host_components" : [{"HostRoles":{"component_name":"ZEPPELIN_MASTER"}}] }' https://$CLUSTER.azurehdinsight.net/api/v1/clusters/$CLUSTER/hosts?Hosts/host_name=$HEADNODE_HOST_FQDN -H "X-Requested-By:ambari"
+
+install
+
+start_service
```

Comment:
we need to at least do some smoke test to ensure the service is started and functional.

Reply:
We ensure that service is in STARTED state